File: 06-prime-check/prime_check.pl
# Perl program to check if a number is prime
# Written by M.V.Harish Kumar on 09-01-2025
use v5.40;

sub is_prime($n) {
    return true if $n == 2;
    do {return false if ($n % $_ == 0)} for 2..(int sqrt $n);
    return $n > 1;
}

say "Prime number checker\n";
print "Enter value for `n`: ";
chomp(my $n = <STDIN>);
say "$n is @{[!is_prime($n) ? 'NOT ' : '']}Prime";

File: 06-prime-check/prime_check.hs
-- Haskell program to check if given number is prime
-- Written by M.V.Harish Kumar on 11-07-2025
import System.IO (hFlush, stdout)

isPrime :: Int -> Bool
isPrime n
    | n < 2 = False
    | n == 2 = True
    | otherwise = all notDiv [2..lim]
    where
        notDiv i = n `rem` i /= 0
        lim = floor . sqrt . fromIntegral $ n

main = do
    putStrLn "Prime number checker\n"
    putStr "Enter a number: "
    hFlush stdout
    n <- getLine
    if isPrime $ read n then
        putStrLn $ n ++ " is a Prime number"
    else
        putStrLn $ n ++ " is NOT a Prime Number"
